ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoSignedTransactionChecker.kt

```diff
@@ -0,0 +1,34 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.ledger.utxo.UtxoLedgerService
+import java.io.Serializable
+
+/**
+ * [UtxoSignedTransactionChecker] verifies a [UtxoSignedTransaction].
+ *
+ * Implement [UtxoSignedTransactionChecker] and pass the implementation into [UtxoLedgerService.receiveFinality] to perform
+ * custom verification on the [UtxoSignedTransaction] received from the initiator of finality.
+ *
+ * When verifying a [UtxoSignedTransaction] throw either an [IllegalArgumentException], [IllegalStateException] or
+ * [CordaRuntimeException] to indicate that the transaction is invalid. This will lead to the termination of finality for the caller of
+ * [UtxoLedgerService.receiveFinality] and all participants included in finalizing the transaction. Other exceptions will still stop
+ * the progression of finality; however, the reason for the failure will not be communicated to the initiator of finality.
+ *
+ * @see UtxoLedgerService.receiveFinality
+ */
+fun interface UtxoSignedTransactionChecker : Serializable {
```

Comment:
Do we need this in the API (is it something that is going to be called by the user)?
I've implemented `UtxoLedgerTransactionVerifier` here: https://github.com/corda/corda-runtime-os/pull/2501/files but it's only part of the runtime because I don't expect users to call it.

@mnesbit  thoughts?

Reply:
Its a callback for users to verify that their transactions are valid to them, rather than the contracts are satisfied. E.g. if you want to check some of the data in the transaction against your database or an external system, that logic would go here.